sklearn/linear_model/least_angle.py

```diff
@@ -132,6 +132,243 @@ def lars_path(X, y, Xy=None, Gram=None, max_iter=500,
+        eps=eps, copy_Gram=copy_Gram, verbose=verbose, return_path=return_path,
+        return_n_iter=return_n_iter, positive=positive)
+
+
+def lars_path_gram(Xy, Gram, n_samples, max_iter=500, alpha_min=0,
+                   method='lar', copy_X=True, eps=np.finfo(np.float).eps,
+                   copy_Gram=True, verbose=0, return_path=True,
+                   return_n_iter=False, positive=False):
+    """lars_path in the sufficient stats mode [1]
+
+    The optimization objective for the case method='lasso' is::
+
+    (1 / (2 * n_samples)) * ||y - Xw||^2_2 + alpha * ||w||_1
+
+    in the case of method='lars', the objective function is only known in
+    the form of an implicit equation (see discussion in [1])
+
+    Read more in the :ref:`User Guide <least_angle_regression>`.
+
+    Parameters
+    -----------
+    Xy : array-like, shape (n_samples,) or (n_samples, n_targets)
+        Xy = np.dot(X.T, y).
+
+    Gram : array, shape: (n_features, n_features)
+        Gram = np.dot(X.T * X).
+
+    n_samples : integer or float
+        Equivalent size of sample.
+
+    max_iter : integer, optional (default=500)
+        Maximum number of iterations to perform, set to infinity for no limit.
+
+    alpha_min : float, optional (default=0)
+        Minimum correlation along the path. It corresponds to the
+        regularization parameter alpha parameter in the Lasso.
+
+    method : {'lar', 'lasso'}, optional (default='lar')
+        Specifies the returned model. Select ``'lar'`` for Least Angle
+        Regression, ``'lasso'`` for the Lasso.
+
+    copy_X : bool, optional (default=True)
+        If ``False``, ``X`` is overwritten.
+
+    eps : float, optional (default=``np.finfo(np.float).eps``)
+        The machine-precision regularization in the computation of the
+        Cholesky diagonal factors. Increase this for very ill-conditioned
+        systems.
+
+    copy_Gram : bool, optional (default=True)
+        If ``False``, ``Gram`` is overwritten.
+
+    verbose : int (default=0)
+        Controls output verbosity.
+
+    return_path : bool, optional (default=True)
+        If ``return_path==True`` returns the entire path, else returns only the
+        last point of the path.
+
+    return_n_iter : bool, optional (default=False)
+        Whether to return the number of iterations.
+
+    positive : boolean (default=False)
+        Restrict coefficients to be >= 0.
+        This option is only allowed with method 'lasso'. Note that the model
+        coefficients will not converge to the ordinary-least-squares solution
+        for small values of alpha. Only coefficients up to the smallest alpha
+        value (``alphas_[alphas_ > 0.].min()`` when fit_path=True) reached by
+        the stepwise Lars-Lasso algorithm are typically in congruence with the
+        solution of the coordinate descent lasso_path function.
+
+    Returns
+    --------
+    alphas : array, shape: [n_alphas + 1]
```

Comment:
`alphas : array, shape (n_alphas + 1,)`

please be consistent with the way you document shapes.